### PR TITLE
feat: add SVG visualization functionality for indexed surfaces

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds the possibility to visualise the `IndexedSurfaces` navigation delegates via the `ACTSVG` plugin.

It also integrates `actsvg` into the proper versioning schema via the master `CMakeLists.txt`file,
and pumps the version to 0.4.33 for some needed extra functionality.

It allows to show the index grid and associations to surfaces that are pointed to:

Screenshot 2023-04-14 at 13 27 29

### CODE REVIEW

**Comment by asalzburger** [comment]
@niermann999 - I selected you as a reviewer, as we will have to repeat this somewhat for `detray` within the context of our summer student project.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (niermann999) [inline Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:3]: Old copyright

File: Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp
Comment (niermann999) [inline Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp:99]: I am not using smart pointers so much, but why is this a ```shared_ptr``` if all that's needed is a memory address?

File: Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp
Comment (niermann999) [inline Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp:124]: Why is this copy needed? ```rSurfacesR0``` does not seem to be used otherwise

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/SurfaceSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/SurfaceSvgConverter.hpp:76]: Sure, fixed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp:106]: Can you open an issue in active? Like this we trace things where they should be.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp:70]: It does. Fixed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp:104]: Fixed.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:71]: This is just some displaying, for this particular test, it sets the text in some nice-to-look-at position.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:38]: Not sure either, but it's pretty explicit like this.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:68]: No, this was just to uniquely name the ```svg``` objects, I changed to ```g``` which will do the trick as well.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:53]: Fixed.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:208]: Fixed.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:139]: Fixed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:132]: Fixed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:39]: Because I am not as clever as you are - fixed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:207]: Renamed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:139]: Leaves you with a `[[maybe_unused]] axesTuple`` so no big win, or?

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:175]: Absolutely, right, I re-ordered and added comments.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:234]: Alright, fixed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:346]: Absolutely and done.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:351]: Uuuuupsi.

Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:351]: Fixed.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp:367]: Added:

```
/// @note this works, because the actual display at the end is 2D
/// in an x-y plane and the parameters are appropriatly defined
///
```

File: Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp:99]: Surfaces in ACTS must be constructed as shared pointers, all other constructors are deleted/private.

File: Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp:124]: Here, we are using the already existing infrastructure of the 'CylindricalSurfaceGeometry' which returns a `const vector` - hence we copy, to add the other surfaces.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:3]: Fixed.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:95]: The grid doesn't really know about the targets, i.e. the draw objects you want to connect, it can draw the tiles etc, but you need to specify what to do with them.

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp]: Fixed.

**Comment by asalzburger** [comment]
Hey @niermann999 - I think I addressed all your comments, the trivial conversions I simply closed, there are only a few for you to follow up then.

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp
Comment (niermann999) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/GridSvgConverter.hpp:96]: Done

File: Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp
Comment (niermann999) [reply Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp:60]: OK, I was just wondering

File: Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp
Comment (niermann999) [reply Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp:99]: Ah, Ok, I didn't know

File: Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp
Comment (niermann999) [reply Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp:124]: Ah, Ok, I could not see the ```const```

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp
Comment (niermann999) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp]: True, and I admit it does not make a big difference. This way, it seemed clearer to me that only the types are needed and the not the values in the tuple. But I don't have a strong opinion on this, I 'll leave it to you :)

**Review by niermann999 — APPROVED** [review]

**Comment by niermann999** [comment]
> Hey @niermann999 - I think I addressed all your comments, the trivial conversions I simply closed, there are only a few for you to follow up then.

Hm, I can't resolve the conversations, but the PR has a green light from my side

**Pull request merged**
